### PR TITLE
[C2] Delete some components that are not used

### PR DESCRIPTION
**Pull request by daniboygg**
This will remove components that are not in use in Vue application. This will delete some unneeded vuex dependencies.

#5377

### CODE REVIEW

**Comment by ywarnier** [comment]
I'm not able to review this one, in particular because of the assets/vue/router/course.js removal. I think *maybe* the corresponding views are a work in progress to be used later. @AngelFQC if these are not works in progress, can we remove them and implement them correctly later on, or should we keep them there to hold the place for the file to be used later?

File: assets/vue/router/course.js [hunk 1]
```diff
@@ -1,29 +0,0 @@
...
-    {
-      name: 'CourseList',
-      path: '',
-      component: () => import('../views/course/List.vue')
-    },
-    {
-      name: 'CourseCreate',
-      path: 'new',
-      component: () => import('../views/course/Create.vue')
-    },
```
Comment (AngelFQC) [inline assets/vue/router/course.js:17]: CourseCreate is currently in use

File: assets/vue/views/course/Create.vue
Comment (AngelFQC) [inline assets/vue/views/course/Create.vue:1]: This page is used to allow teachers to create their courses  (from the orange button on the /courses page)

**Review by AngelFQC — CHANGES_REQUESTED** [review]

File: assets/vue/views/course/Create.vue [hunk 2]
```diff
@@ -31,18 +30,17 @@ import courseService from "../../services/courseService"
...
-const store = useStore()
 const item = ref({})
 const router = useRouter()
 const { t } = useI18n()
 
-const isLoading = computed(() => store.getters['course/getField']('isLoading'))
-const violations = computed(() => store.getters['course/getField']('violations'))
-const courseData = ref({})
+const isLoading = ref(false)
+const violations = ref(null)
```
Comment (daniboygg) [inline assets/vue/views/course/Create.vue:38]: The reference violations is used in the component CourseForm but this component has no property errors...
```html
<CourseForm
      ref="createForm"
      :errors="violations"
      :values="item"
      @submit="submitCourse"
    />
```

**Comment by daniboygg** [comment]
I reverted the changes. So, do we need the components CourseList, CourseUpdate and CourseShow. All of them are using vuex, but when I visit the route I see nothing. 
I think we should fix these components before remove vuex, otherwise it will be difficult to ensure the behavior is consistent with the old one.

**Comment by ywarnier** [comment]
Tested. I don't see anything missing, feature-wise, by testing it with this PR. Will include soon in master.

**Pull request merged**